pkg/common/cns-lib/volume/util.go

```diff
@@ -66,11 +75,49 @@ func IsDiskAttached(ctx context.Context, vm *cnsvsphere.VirtualMachine, volumeID
 	return "", nil
 }
 
+func hasNVMEController(ctx context.Context, devices object.VirtualDeviceList) bool {
+	log := logger.GetLogger(ctx)
+	_, err := devices.FindNVMEController("")
+	if err != nil {
+		log.Debugf("Could not find any NVMEController")
+		return false
+	}
+	return true
+}
+
+// getNvmeUUID returns the NVME formatted UUID
+func getNvmeUUID(ctx context.Context, uuid string) (string, error) {
+	log := logger.GetLogger(ctx)
+	uuidBytes, err := uuidlib.Parse(uuid)
+	if err != nil {
+		log.Errorf("Error while parsing uuid with err=%v", err)
+		return "", err
+	}
+	var nvmeUUID uuidlib.UUID
+	nvmeUUID[0] = uuidBytes[8]
```

Comment:
Is the difference little endian vs big endian ? Is there a golang package that can do that conversion?

Reply:
> Is there a golang package that can do that conversion?
It would be better to look for packages only from wellknown places (e.g. golang standard libraries or k8s packages).
Minimizing external dependency will make your code simpler.

Reply:
Let me take this an action item to see if there are well known packages to replace the conversion logic

Reply:
I will follow-up with this suggestion in the next iteration. Created an issue to track the same: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/913